src/main/java/pl/nogacz/checkers/board/Board.java

```diff
@@ -136,6 +144,245 @@ public void readKeyboard(KeyEvent event) {
+                        Design.addPawn(coord, pawn);
+                        board.put(coord, pawn);
+                    }
+        
+                    for(int i = 0; i < blackQueenCoords.size(); i++) {           
+                        int x = Integer.parseInt(blackQueenCoords.get(i).charAt(0)+"");  
+                        int y = Integer.parseInt(blackQueenCoords.get(i).charAt(2)+"");  
+                        Coordinates coord = new Coordinates(x,y);
+                        PawnClass pawn = new PawnClass(Pawn.QUEEN, PawnColor.BLACK);
+                        Design.removePawn(coord);
+                        board.remove(coord);
+                        Design.addPawn(coord, pawn);
+                        board.put(coord, pawn);
+                    }  
+                    System.out.println("You loaded a game!");                         
+                }
+                catch(IOException e){
+                    System.out.println("There is no saved game!");
+                }
+                alert.close();
+            }
+            else if (result.get() == loadFromHome) {
+                String userHomeFolder = System.getProperty("user.home");
+                File file = new File(userHomeFolder, "JavaCheckersSavedGame.txt");
+                try(BufferedReader load = new BufferedReader(new FileReader(file))) {
+                    ArrayList<String> blackCoords = new ArrayList<String>();
+                    ArrayList<String> whiteCoords = new ArrayList<String>();
+                    ArrayList<String> blackQueenCoords = new ArrayList<String>();
+                    ArrayList<String> whiteQueenCoords = new ArrayList<String>();
+                    int white_count = Integer.parseInt(load.readLine());  
+                    for(int i = 0; i < white_count; i++)
+                        whiteCoords.add(load.readLine());
+                    int black_count = Integer.parseInt(load.readLine());     
+                    for(int i = 0; i < black_count; i++)
+                        blackCoords.add(load.readLine());     
+                    int whitequeen_count = Integer.parseInt(load.readLine());  
+                    for(int i = 0; i < whitequeen_count; i++)
+                        whiteQueenCoords.add(load.readLine());
+                    int blackqueen_count = Integer.parseInt(load.readLine());     
+                    for(int i = 0; i < blackqueen_count; i++)
+                        blackQueenCoords.add(load.readLine());                        
+                    load.close();          
+                    for(int x = 0; x <= 7; x++) {
+                        for(int y = 0; y <= 7; y++) {
+                            Coordinates coord = new Coordinates(x,y);
+                            Design.removePawn(coord);
+                            board.remove(coord);
+                        }
+                    }
+                    for(int i = 0; i < whiteCoords.size(); i++) {           

```

Comment:
Benzer kod tekrarları bu for döngülerde de bar. Yukarıdaki yazdığıma benzer bir yaklalımla kodları daha sade hale getir.